app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java

```diff
@@ -331,24 +328,15 @@ public void updateDrawState(TextPaint tp) {
                         tp.baselineShift -= (int) (relativeVerticalShiftRatio * tp.getFontMetrics().top);
                     }
                 }
-
-                // Ratio values tested on Android 13, Samsung, Google and OnePlus branded phones, using screen densities of 300 to 560
-                // On other devices and fonts the left separator may be vertically shifted by a few pixels,
-                // but it's good enough and still visually better than not doing this scaling/shifting
-                final float verticalShiftRatio = -0.38f; // shift up by 38%
-                final float verticalLeftSeparatorShiftRatio = -0.075f; // shift up by 8%
-                final float horizontalStretchRatio = 0.92f; // stretch narrower by 8%
-                final float leftSeparatorFontRatio = 1.87f;  // increase height by 87%
-
-                addSpanStyling(leftSeparatorSpan, new RelativeSizeSpan(leftSeparatorFontRatio));
-                addSpanStyling(leftSeparatorSpan, new ScaleXSpan(horizontalStretchRatio));
-
-                // shift the left separator up by a smaller amount, to visually align it after changing the size
-                addSpanStyling(leftSeparatorSpan, new RelativeVerticalOffsetSpan(verticalLeftSeparatorShiftRatio));
-                addSpanStyling(likesSpan, new RelativeVerticalOffsetSpan(verticalShiftRatio));
-                addSpanStyling(middleSeparatorSpan, new RelativeVerticalOffsetSpan(verticalShiftRatio));
-                addSpanStyling(dislikeSpan, new RelativeVerticalOffsetSpan(verticalShiftRatio));
-
+                // shift everything up, to compensate for the vertical movement caused by the font change below
+                addSpanStyling(leftSeparatorSpan, new RelativeVerticalOffsetSpan(segmentedVerticalShiftRatio));
```

Comment:
Is it really necessary to instantiate the class every time `updateDislike` is called?

Reply:
Instantiate the style spans?  Maybe they could be reused, not sure though.  But they are not heavy weight objects or expensive to create.

Reply:
I tried using a single instance of RelativeVericalOffsetSpan across all of the different spans (like, dislike, separators), but the alignment was wrong.  Using individual offset spans fixed that.

Reply:
Just tried again to confirm, and each span needs it's own instance.  (otherwise the sections don't like up vertically)